src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs

```diff
@@ -327,10 +327,19 @@ internal static void ApplyParameterInfo(this JsonNode schema, ApiParameterDescri
     /// <param name="createSchemaReferenceId">A delegate that generates the reference ID to create for a type.</param>
     internal static void ApplyPolymorphismOptions(this JsonNode schema, JsonSchemaExporterContext context, Func<JsonTypeInfo, string?> createSchemaReferenceId)
     {
-        // The `context.Path.Length == 0` check is used to ensure that we only apply the polymorphism options
+        // The `context.BaseTypeInfo == null` check is used to ensure that we only apply the polymorphism options
```

Comment:
Does schema nesting represent subtyping?  I would have guessed it was for composition.  Why would having no base type (I'm assuming there's no "top" type like `object`) correspond to having no containing schema?

Reply:
`BaseTypeInfo` is strictly related with polymorphic subtyping in this case (see the implementation of the API [here](https://github.com/dotnet/runtime/commit/7f5ce1e2bb5b7589281a713dc379671cc0b99ea6) for some additional insight). In this case, the `BaseTypeInfo` is only set if we're generating the subschema for a derived type.

Reply:
"Subschema" in the sense that it's one of several in a discriminated union?

Reply:
As discussed offline, type hierarchies are represented as discriminated unions and subtypes are subschemas in the sense that they are nested within the discriminated union schema.